### PR TITLE
Less Cry Legacy

### PR DESCRIPTION
**Pull request by nemerle**
This is mostly a draft/RFC to gather comments about the removals 

Quick note- some of the changes are 'stale', I'll try to clean those up as I do another pass over things :smile:

### CODE REVIEW

File: Code/Editor/Objects/BaseObject.h [hunk 1]
```diff
@@ -548,52 +483,23 @@ class SANDBOX_API CBaseObject
...
-    // Procedural Floor Management.
-    //////////////////////////////////////////////////////////////////////////
-    int GetFloorNumber() const { return m_floorNumber; };
-    void SetFloorNumber(int floorNumber) { m_floorNumber = floorNumber; };
-
-    virtual void OnPropertyChanged(IVariable*);
-    virtual void OnMultiSelPropertyChanged(IVariable*);
-
-    //! Draw a reddish highlight indicating its budget usage.
+        //! Draw a reddish highlight indicating its budget usage.
```
Comment (bosnichd) [inline Code/Editor/Objects/BaseObject.h:495]: Random whitespace introduced.

File: Code/Legacy/CryCommon/IFunctorBase.h [hunk 2]
```diff
@@ -8,47 +8,4 @@
 
 
 // Description : Base header for multi DLL functors.
-
-
-#ifndef CRYINCLUDE_CRYCOMMON_IFUNCTORBASE_H
-#define CRYINCLUDE_CRYCOMMON_IFUNCTORBASE_H
 #pragma once
```
Comment (bosnichd) [inline Code/Legacy/CryCommon/IFunctorBase.h:11]: Can we remove this entire file?

File: Code/Framework/AzFramework/Platform/Android/AzFramework/Archive/ArchiveVars_Android.h [hunk 3]
```diff
@@ -7,5 +7,3 @@
  */
 
 #pragma once
```
Comment (bosnichd) [inline Code/Framework/AzFramework/Platform/Android/AzFramework/Archive/ArchiveVars_Android.h:9]: Can we remove this entire file?

File: Code/Framework/AzFramework/Platform/Linux/AzFramework/Archive/ArchiveVars_Linux.h [hunk 4]
```diff
@@ -7,5 +7,3 @@
  */
 
 #pragma once
```
Comment (bosnichd) [inline Code/Framework/AzFramework/Platform/Linux/AzFramework/Archive/ArchiveVars_Linux.h:9]: Can we remove this entire file?

File: Code/Framework/AzFramework/Platform/Mac/AzFramework/Archive/ArchiveVars_Mac.h [hunk 5]
```diff
@@ -7,5 +7,3 @@
  */
 
 #pragma once
```
Comment (bosnichd) [inline Code/Framework/AzFramework/Platform/Mac/AzFramework/Archive/ArchiveVars_Mac.h:9]: Can we remove this entire file?

File: Code/Framework/AzFramework/Platform/Windows/AzFramework/Archive/ArchiveVars_Windows.h [hunk 6]
```diff
@@ -7,5 +7,3 @@
  */
 
 #pragma once
```
Comment (bosnichd) [inline Code/Framework/AzFramework/Platform/Windows/AzFramework/Archive/ArchiveVars_Windows.h:9]: Can we remove this entire file?

File: Code/Framework/AzFramework/Platform/iOS/AzFramework/Archive/ArchiveVars_iOS.h [hunk 7]
```diff
@@ -7,5 +7,3 @@
  */
 
 #pragma once
```
Comment (bosnichd) [inline Code/Framework/AzFramework/Platform/iOS/AzFramework/Archive/ArchiveVars_iOS.h:9]: Can we remove this entire file?

**Review by bosnichd — APPROVED** [review]
Nice, thanks!

File: Code/Framework/AzFramework/Platform/iOS/AzFramework/Archive/ArchiveVars_iOS.h (hunk 7 above)
Comment (nemerle) [reply Code/Framework/AzFramework/Platform/iOS/AzFramework/Archive/ArchiveVars_iOS.h:9]: Likely, I'm currently preparing a 'clean' version of this PR, will take a look in there :)

**Comment by nemerle** [comment]
Superseded by #5961

**Pull request closed without merging**